tests/unit/ui/components/TestShowMoreCard.js

```diff
@@ -1,68 +1,84 @@
 import * as React from 'react';
-import { mount } from 'enzyme';
-import { findDOMNode } from 'react-dom';
-import { Simulate, renderIntoDocument } from 'react-dom/test-utils';
-
-import { ShowMoreCardBase, MAX_HEIGHT } from 'ui/components/ShowMoreCard';
-import { fakeI18n } from 'tests/unit/helpers';
-
-function render(props) {
-  return renderIntoDocument(<ShowMoreCardBase i18n={fakeI18n()} {...props} />);
-}
+import { shallow, mount } from 'enzyme';
+
+import ShowMoreCard, {
+  ShowMoreCardBase,
+  MAX_HEIGHT,
+} from 'ui/components/ShowMoreCard';
+import {
+  createFakeEvent,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
 
 describe(__filename, () => {
+  const render = (props) => {
+    return shallowUntilTarget(
+      <ShowMoreCard i18n={fakeI18n()} {...props} />,
+      ShowMoreCardBase,
+    );
+  };
+
   it('reveals more text when clicking "show more" link', () => {
     const root = render({ children: 'Hello I am description' });
-    const rootNode = findDOMNode(root);
 
-    // We have to manually set the expanded flag to false because we
-    // don't have a clientHeight in the tests.
+    // We have to manually set the expanded flag to false because we don't have
+    // a clientHeight in the tests.
     root.setState({ expanded: false });
-    expect(rootNode.className).not.toContain('.ShowMoreCard--expanded');
-    expect(root.state.expanded).toEqual(false);
-    expect(rootNode.querySelector('.Card-footer-link').textContent).toEqual(
-      'Expand to Read more',
+
+    expect(root).not.toHaveClassName('ShowMoreCard--expanded');
+    expect(root).toHaveState('expanded', false);
+
+    // We have to shallow this prop because we pass a component to it.
+    expect(shallow(root.prop('footerLink')).html()).toContain(
+      '<span class="visually-hidden">Expand to</span> Read more',
     );
 
-    Simulate.click(rootNode.querySelector('.Card-footer-link a'));
+    // We cannot directly target the footer link, so we simulate the click here.
+    root.instance().onClick(createFakeEvent());
+    root.update();
 
-    expect(rootNode.className).toContain('ShowMoreCard--expanded');
-    expect(root.state.expanded).toEqual(true);
+    expect(root).toHaveClassName('ShowMoreCard--expanded');
+    expect(root).toHaveState('expanded', true);
 
-    expect(rootNode.querySelector('.Card-footer-link')).toEqual(null);
+    expect(root).toHaveProp('footerLink', null);
   });
 
   it('is expanded by default', () => {
     const root = render({ children: 'Hello I am description' });
-    expect(root.state.expanded).toEqual(true);
+
+    expect(root).toHaveState('expanded', true);
```

Comment:
should we check the behavior (for existing class). I guess I was the impression that was better..let me know if I am missing something tho :/